02 Sample App/01 Login/src/pages/login/pageContainer.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import * as toastr from 'toastr';
+import {loginAPI} from '../../rest-api/login/loginAPI';
+import {LoginCredential} from '../../models/loginCredential';
+import {UserProfile} from '../../models/userProfile';
+import {LoginPage} from './page';
+
+interface State {
+  loginCredential: LoginCredential;
+}
+
+export class LoginPageContainer extends React.Component <{}, State> {
+  constructor() {
+    super();
+
+    this.state = {
+      loginCredential: new LoginCredential(),
+    };
+  }
+
+  private updateLoginInfo(fieldName: string, value: string) {
+    this.setState({
+      loginCredential: {
+        ...this.state.loginCredential,
+        [fieldName]: value,
+      }
+    });
```

Comment:
I would just add a comment indicating the old fahsioned way to do this, and explain a bit about the new way and add a link where the sutdent can get more info about the "..." operator